apps/web/app/composables/useProduct/useProduct.ts

```diff
@@ -47,9 +48,9 @@ export const useProduct: UseProductReturn = (slug) => {
       () => useSdk().plentysystems.getProduct(params),
     );
     useHandleError(error.value ?? null);
-    await setupBlocks(
-      (state.value.data.blocks?.length ? state.value.data.blocks : useProductTemplateData()) as Block[],
-    );
+
+    const dbBlocks = data.value?.data.blocks;
+    await setupBlocks((dbBlocks && dbBlocks.length > 0 ? dbBlocks : useProductTemplateData()) as Block[]);
```

Comment:
```suggestion
    const fetchedBlocks = data.value?.data.blocks;
    await setupBlocks((fetchedBlocks && fetchedBlocks.length > 0 ? dbBlocks : useProductTemplateData()) as Block[]);
```
Let's stick to `fetch` as our terminology here.